system-addon/lib/TopSitesFeed.jsm

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ /* globals PlacesProvider, PreviewProvider, Task */
+"use strict";
+
+const {utils: Cu} = Components;
+const {actionTypes: at, actionCreators: ac} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+
+Cu.import("resource:///modules/PlacesProvider.jsm");
+Cu.import("resource:///modules/PreviewProvider.jsm");
+Cu.import("resource://gre/modules/Task.jsm");
+
+this.TopSitesFeed = class TopSitesFeed {
+  getData(action) {
+    return Task.spawn(function*() {
+      let newAction;
+      let links = yield PlacesProvider.links.getLinks();
+      let result;
+      if (!links) {
+        result = [];
+      } else {
+        result = links.filter(link => link && link.type !== "affiliate").slice(0, 12);
+      }
+      newAction = {type: at.TOP_SITES_UPDATED, data: result};
+      this.store.dispatch(ac.SendToContent(newAction, action.meta.fromTarget));
```

Comment:
Add a note here about why this needs to be sent to content immediately because of preloading, and that it should be changed to a regular dispatch if we don't use NEW_TAB_LOAD as the trigger for updating